invisible_cities/detector_simulation/buffer_functions.py

```diff
@@ -0,0 +1,143 @@
+                t_min     : float       ,
+                t_max     : float       ,
+                max_buffer: int         ) -> Tuple[np.ndarray, pd.Series]:
+    """
+    Raw data binning function.
+
+    Parameters
+    ----------
+    sensors : List of Waveforms for one sensor type
+    Should be sorted into one type/binning
+    t_min   : float
+              Minimum time to be used to define bins.
+    t_max   : float
+              As t_min but the maximum to be used
+    """
+    max_time    = min(t_max, t_min + max_buffer)
+    min_bin     = np.floor(t_min    / bin_width) * bin_width
+    max_bin     = np.ceil (max_time / bin_width) * bin_width
+
+    bins        = np.arange(min_bin, max_bin + bin_width, bin_width)
+    bin_sensors = sensors.groupby('sensor_id').apply(weighted_histogram, bins)
+    return bins[:-1], bin_sensors
+
+
+## !! to-do: clarify for non-pmt versions of next
+## !! to-do: Check on integral instead of only threshold?
+def find_signal_start(wfs          : pd.Series,
+                      bin_threshold: float    ,
+                      stand_off    : int      ) -> List[int]:
+    """
+    Finds signal in the binned waveforms and
+    identifies candidate triggers.
+    """
+    eng_sum = wfs.sum()
+    indices = indices_and_wf_above_threshold(eng_sum,
+                                             bin_threshold).indices
+    ## Just using this and the stand_off for now
+    ## taking first above sum threshold.
+    ## !! To-do: make more robust with min int? or similar
+    all_indx = split_in_peaks(indices, stand_off)
+    return [pulse[0] for pulse in all_indx]
+
+
+def pad_safe(sensors: np.ndarray, padding: Tuple) -> np.ndarray:
+    """Pads zeros around each sensor in a 2D array"""
+    if not sensors.shape[0]:
+        return np.empty((0, padding[0] + padding[1] + 1))
+    return np.apply_along_axis(np.pad, 1, sensors, padding, "constant")
+
+
+def buffer_calculator(buffer_len: float, pre_trigger: float,
+                      pmt_binwid: float, sipm_binwid: float) -> Callable:
+    """
+    Calculates the output buffers for all sensors
+    based on a configured buffer length and pretrigger.
+    Synchronising the clock between the two sensor types.
+
+    Parameters
+    ----------
+    buffer_len  : float
+                  Length of buffer expected in mus
+    pre_trigger : float
+                  Time in buffer before identified signal in mus
+    pmt_binwid  : float
+                  Width in mus of PMT sample integration
+    sipm_binwid : float
+                  Width in mus of SiPM sample integration
+    """
+    pmt_buffer_samples  = int(buffer_len /  pmt_binwid)
+    sipm_buffer_samples = int(buffer_len / sipm_binwid)
+    sipm_pretrg         = int(pre_trigger / sipm_binwid)
+    sipm_postrg         = sipm_buffer_samples - sipm_pretrg
+    pmt_pretrg_base     = int(pre_trigger / pmt_binwid)
+    pmt_postrg_base     = pmt_buffer_samples - pmt_pretrg_base
```

Comment:
```suggestion
    pmt_buffer_samples  = int(buffer_len  //  pmt_binwid)
    sipm_buffer_samples = int(buffer_len  // sipm_binwid)
    sipm_pretrg         = int(pre_trigger // sipm_binwid)
    sipm_postrg         = sipm_buffer_samples - sipm_pretrg
    pmt_pretrg_base     = int(pre_trigger  // pmt_binwid)
    pmt_postrg_base     = pmt_buffer_samples - pmt_pretrg_base
```
technically integer division is more appropriate, although there is no effective change due to int conversion